CMakeLists.txt

```diff
@@ -28,8 +28,24 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
 # Unused language features.
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-rtti -fno-exceptions")
 
-#set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fsanitize=address")
-#set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=address")
+if (CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  # Global options (these apply for other subprojects like JSObjects) and must
+  # be set before we include subdirectories.
+  if (SANITIZER)
+    set(COMMON_FLAGS "${COMMON_FLAGS} -g -fno-omit-frame-pointer")
+  endif ()
+
+  if (SANITIZER STREQUAL "asan")
```

Comment:
We do it this way because "asan" is the name people know about, and because for other sanitizers (e.g.: ubsan), we can't just take the cmake argument and pass it down.